automerge-wasm/test/test.ts

```diff
@@ -538,6 +538,31 @@ describe('Automerge', () => {
       assert.deepStrictEqual(doc.save(), doc2.save())
     })
 
+    it('should handle merging text conflicts then saving & loading', () => {
+      let A = create()
+      let At = A.make('_root', 'text', TEXT)
+      A.splice(At, 0, 0, Array.from('hello'))
+
+      let B = A.clone()
+      let Bt = B.value('_root', 'text')
+      if (!Bt || Bt[0] !== 'text') return assert.fail()
+      let obj = Bt[1]
+      B.splice(obj, 4, 1, '')
+      B.splice(obj, 4, 0, '!')
+      B.splice(obj, 5, 0, ' ')
+      B.splice(obj, 6, 0, Array.from('world'))
+
+      A.applyChanges(B.getChanges(A.getHeads()))
+
+      let binary = A.save()
+
+      let C = loadDoc(binary)
+
+      assert.deepEqual(C.value('_root', 'text'), ['text', 'hello world'])
```

Comment:
Would be nice to use the `assert_doc!` helper macro from the tests in the `automerge` crate here. But probably we would need to do some work to make that accessible outside the test crate so probably something to do another day.